internal/services/cdn/cdn_endpoint_custom_domain_resource.go

```diff
@@ -32,9 +40,10 @@ func resourceArmCdnEndpointCustomDomain() *pluginsdk.Resource {
 		}),
 
 		Timeouts: &pluginsdk.ResourceTimeout{
-			Create: pluginsdk.DefaultTimeout(30 * time.Minute),
+			Create: pluginsdk.DefaultTimeout(20 * time.Hour),
 			Read:   pluginsdk.DefaultTimeout(5 * time.Minute),
-			Delete: pluginsdk.DefaultTimeout(30 * time.Minute),
+			Update: pluginsdk.DefaultTimeout(20 * time.Hour),
+			Delete: pluginsdk.DefaultTimeout(20 * time.Hour),
```

Comment:
Yes, the CDN managed HTTPS custom domain deployment takes a long time. See: https://docs.microsoft.com/en-us/azure/cdn/cdn-custom-ssl?tabs=option-1-default-enable-https-with-a-cdn-managed-certificate#wait-for-propagation. Though the document stated it takes 6-8 hours, while in fact it sometimes exceeds that limit, therefore I set a relatively large interval here in case.